src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5342,6 +5355,63 @@ static BoundNode bindSpreadElement(SpreadElementSyntax syntax, BindingDiagnostic
+                // we still bind the arguments in those later with elements.  However, we only validate those
+                // arguments against the final arguments against the destination target type if the with element
+                // was in the proper position.
+
+                var analyzedArguments = AnalyzedArguments.GetInstance();
+
+                // PROTOTYPE: Spec says we should only allow arglist if trivial.  Circle back on this and see if
+                // this just falls out with the 'allowArgList: true' below.  If so, let LDM know it was easy and
+                // allow it.  If it requires substantial work beyond this, disallow it for this feature.
+                BindArgumentsAndNames(withElementSyntax.ArgumentList, diagnostics, analyzedArguments, allowArglist: true);
+
+                var arguments = analyzedArguments.Arguments;
+                for (int i = 0; i < arguments.Count; i++)
+                {
+                    var arg = arguments[i];
+                    if (arg.Type is { TypeKind: TypeKind.Dynamic })
+                    {
+                        diagnostics.Add(ErrorCode.ERR_CollectionArgumentsDynamicBinding, arg.Syntax);
+                        arguments[i] = new BoundBadExpression(arg.Syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [arg], type: Compilation.GetSpecialType(SpecialType.System_Object));
+                    }
+                }
+
+                BoundUnconvertedWithElement? withElement;
+                BoundBadExpression? badExpression;
+
+                if (withElementSyntax == syntax.Elements.First())
+                {
+                    // Got a with-element, and it was in the right place.  Pass it along directly in
+                    // unconverted-collection-expression so that we can construct the collectin properly.
+                    withElement = new BoundUnconvertedWithElement(
+                        withElementSyntax,
+                        analyzedArguments.Arguments.ToImmutable(),
+                        analyzedArguments.Names.ToImmutableOrNull(),
+                        analyzedArguments.RefKinds.ToImmutableOrNull());
+                    badExpression = null;
+                }
+                else
+                {
+                    // Improperly placed with-element.  Report an error and pass along the arguments so they remain
+                    // in the tree for further analysis, but replace the with-element itself with a bad node so that
+                    // it doesn't influence later transformations.
+                    diagnostics.Add(ErrorCode.ERR_CollectionArgumentsMustBeFirst, withElementSyntax.WithKeyword);
+
+                    withElement = null;
+                    badExpression = MakeBadExpressionForObjectCreation(
+                        withElementSyntax, this.Compilation.GetSpecialType(SpecialType.System_Object),
```

Comment:
>System_Object [](http://example.com/codeflow?start=87&length=13)

The type `object` is wrong and I'm not sure the benefit of using `MakeBadExpressionForObjectCreation` helper.
Consider `BadExpression(node, BuildArgumentsForErrorRecovery(analyzedArguments)`.

Reply:
nice!